pkg/gcc/loss_based_bwe.go

```diff
@@ -0,0 +1,93 @@
+package gcc
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"github.com/pion/logging"
+)
+
+// LossStats contains internal statistics of the loss based controller
+type LossStats struct {
+	TargetBitrate int
+	AverageLoss   float64
+}
+
+type lossBasedBandwidthEstimator struct {
+	lock           sync.Mutex
+	maxBitrate     int
+	minBitrate     int
+	bitrate        int
+	averageLoss    float64
+	lastLossUpdate time.Time
+	lastIncrease   time.Time
+	lastDecrease   time.Time
+	log            logging.LeveledLogger
+}
+
+func newLossBasedBWE(initialBitrate int) *lossBasedBandwidthEstimator {
+	return &lossBasedBandwidthEstimator{
+		lock:           sync.Mutex{},
+		maxBitrate:     100_000_000, // 100 mbit
+		minBitrate:     100_000,     // 100 kbit
+		bitrate:        initialBitrate,
+		averageLoss:    0,
+		lastLossUpdate: time.Time{},
+		lastIncrease:   time.Time{},
+		lastDecrease:   time.Time{},
+		log:            logging.NewDefaultLoggerFactory().NewLogger("gcc_loss_controller"),
+	}
+}
+
+func (e *lossBasedBandwidthEstimator) getEstimate(wantedRate int) LossStats {
+	e.lock.Lock()
+	defer e.lock.Unlock()
+
+	if e.bitrate <= 0 {
+		e.bitrate = clampInt(wantedRate, e.minBitrate, e.maxBitrate)
+	}
+	e.bitrate = minInt(wantedRate, e.bitrate)
+
+	return LossStats{
+		TargetBitrate: e.bitrate,
+		AverageLoss:   e.averageLoss,
+	}
+}
+
+func (e *lossBasedBandwidthEstimator) updateLossEstimate(results []Acknowledgment) {
+	if len(results) == 0 {
+		return
+	}
+
+	packetsLost := 0
+	for _, p := range results {
+		if p.Arrival.IsZero() {
+			packetsLost++
+		}
+	}
+
+	lossRatio := float64(packetsLost) / float64(len(results))
+	e.averageLoss = e.average(time.Since(e.lastLossUpdate), e.averageLoss, lossRatio)
+	e.lastLossUpdate = time.Now()
+
+	increaseLoss := math.Max(e.averageLoss, lossRatio)
+	decreaseLoss := math.Min(e.averageLoss, lossRatio)
+
+	e.lock.Lock()
+	defer e.lock.Unlock()
+
+	if increaseLoss < 0.02 && time.Since(e.lastIncrease) > 200*time.Millisecond {
```

Comment:
I would make the thresholds + time constants `0.02` and `200 * time.Millisecond`  Also might be worth linking the PDF I remember seeing the constants in there?